### ISSUE TITLE
Incorrect parsing due to locale settings

### ISSUE THREAD

**Issue by noskill** [opencog/link-grammar#804]
Following program return different result depending on locale settings:

```
#include <locale.h>
#include <stdio.h>
#include "link-includes.h"


int main()
{
    Dictionary    dict;
    Parse_Options opts;
    Sentence      sent;
    Linkage       linkage;
    char *        diagram;
    int           i, num_linkages;
    const char * sentence = "Is the bed white?";
    setlocale(LC_ALL, "");
    opts = parse_options_create();
    dict = dictionary_create_default_lang();
    if (!dict) {
        printf ("Fatal error: Unable to open the dictionary\n");
        return 1;
    }

    prt_error("Info: Dictionary version %s, locale %s\n",
            linkgrammar_get_dict_version(dict),
            linkgrammar_get_dict_locale(dict));
    prt_error("Info: Library version %s. Enter \"!help\" for help.\n",
            linkgrammar_get_version());

 
    sent = sentence_create(sentence, dict);

                        parse_options_set_min_null_count(opts, 0);
                        parse_options_set_max_null_count(opts, 0);
                        parse_options_reset_resources(opts);

    
    num_linkages = sentence_parse(sent, opts);
    printf("Num linkages %i found\n", num_linkages);
    for(int i=0; i<num_linkages; i++)
    {
        linkage = linkage_create(i, sent, opts);
        printf("%s\n", diagram = linkage_print_diagram(linkage, true, 800));
        fprintf(stdout, "cost vector = (UNUSED=%d DIS=%5.2f LEN=%d)\n",
                                linkage_unused_word_cost(linkage),
                                linkage_disjunct_cost(linkage),
                                linkage_link_cost(linkage));

        linkage_free_diagram(diagram);
        linkage_delete(linkage);
    }
    sentence_delete(sent);

    dictionary_delete(dict);
    parse_options_delete(opts);
    return 0;
}
```
correct parse:

```
LC_NUMERIC=C ./a.out | head
link-grammar: Info: Dictionary found at ./data/en/4.0.dict
link-grammar: Info: Dictionary version 5.5.2, locale en_US.UTF-8
link-grammar: Info: Library version link-grammar-5.5.1. Enter "!help" for help.
link-grammar: Info: Freeing dictionary en/4.0.dict
link-grammar: Info: Freeing dictionary en/4.0.affix
Num linkages 12 found

    +--------------Xp-------------+
    |       +-------Pa-------+    |
    |       +--SIs*x--+      |    |
    +-->Qd--+   +Ds**c+      |    +--RW--+
    |       |   |     |      |    |      |
LEFT-WALL is.v the  bed.n white.a ? RIGHT-WALL
```

wrong parse:

```
 LC_NUMERIC=ru_RU.utf-8 ./a.out | head
link-grammar: Info: Dictionary found at ./data/en/4.0.dict
link-grammar: Info: Dictionary version 5.5.2, locale en_US.UTF-8
link-grammar: Info: Library version link-grammar-5.5.1. Enter "!help" for help.
link-grammar: Info: Freeing dictionary en/4.0.dict
link-grammar: Info: Freeing dictionary en/4.0.affix
Num linkages 12 found

    +--------------Xp-------------+
    |       +-------I*v------+    |
    |       +--SIs*b--+      |    |
    +-->Qd--+   +Ds**c+      |    +--RW--+
    |       |   |     |      |    |      |
LEFT-WALL is.v the  bed.n white.v ? RIGHT-WALL

```

**Comment by linas** [comment]
Given that there's this short, stand-alone main, maybe we can convert this into a unit test?  Would that be hard?

**Comment by ampli** [comment]
Unit test of correct cost conversion using this example sentence?
Or can we just add this sentence/locale to the Python tests?

**Comment by linas** [comment]
Don't much care right now. Adding this sentence/locale to python is probably enough.  But what if tester does not have the `ru` locale installed? Ughhhh.

**Comment by ampli** [comment]
> But what if tester does not have the `ru` locale installed?

I propose to skip the test in that case (with a reason message when using -v).
(The same can be done for the current `test_dictionary_locale_definition` that uses the `tr` locale.)

**Comment by linas** [comment]
that works ...

**Comment by vsbogd** [comment]
Just found that ```en_DK.utf8``` locale also has ```decimal_point == ','```. And on my system this locale is installed by default, so probably it can replace ```ru_RU.utf8``` as more frequent:
```
$ for i in `locale -a`; do echo -n "$i => " ; LC_NUMERIC=$i locale decimal_point ; done | grep '=> ,'
en_DK.utf8 => ,
ru_RU.utf8 => ,
```

**Comment by ampli** [comment]
Thanks for reporting this problem. It has been fixed in PRs #805 and #809.
I'm closing this issue. Please reopen it if there is still such a problem after this fix.

**Comment by linas** [comment]
there should probably (?) be a changelog entry for this fix.
